packages/rum-core/src/domain/rumEventsCollection/action/getSelectorFromElement.ts

```diff
@@ -0,0 +1,52 @@
+import { objectValues } from '@datadog/browser-core'
+
+type SelectorStrategy = (
+  element: Element | HTMLElement | HTMLInputElement | HTMLSelectElement
+) => string | undefined | null
+
+export function getSelectorFromElement(targetElement: Element): string {
+  const strategies: { [name: string]: SelectorStrategy } = {
+    getIDSelector: (element) => {
+      const isUnique = element.id && element.ownerDocument.body.querySelectorAll(`#${element.id}`).length === 1
+
+      if (isUnique) return `#${element.id}`
+    },
+    getClassSelector: (element) => {
+      const orderedClassList = Array.from(element.classList).sort()
+      const siblings = Array.from(element.parentElement!.children).filter((child) => child !== element)
+      const classUniqueAmongSiblings = siblings.every(
+        (sibling) =>
+          sibling.tagName !== element.tagName ||
+          orderedClassList.some((className) => !sibling.classList.contains(className))
+      )
+
+      if (classUniqueAmongSiblings)
+        return `${element.tagName}${orderedClassList.map((className) => `.${className}`).join('')}`
+    },
+    getPositionSelector: (element) => {
+      const siblings = element.parentElement!.children
+      const isUniqueChild = siblings.length === 1
+      let index = 0
+      while (element === siblings.item(index)) index++
+
+      return `${element.tagName}${!isUniqueChild ? `:nth-of-type(${index + 1})` : ''}`
+    },
+  }
+
+  const targetElementSelector = []
+  let element: Element | null = targetElement
+  let strategy
+
+  while (element && element.nodeName !== 'HTML' && strategy !== strategies.getIDSelector) {
+    for (strategy of objectValues(strategies)) {
+      const elementSelector = strategy(element)
+      if (elementSelector) {
+        targetElementSelector.unshift(elementSelector)
+        break
+      }
+    }
+    element = element.parentElement
+  }
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** It could be interesting to compare the implementation with a simpler approach using the strategies directly instead of iterating over an array of strategies. 
IIUC, we don't plan to add a bunch of other strategies so it could just be more explicit.

Reply:
Yes sure